lib/pacio_test_kit/common_tests/search_type_test.rb

```diff
@@ -0,0 +1,40 @@
+require_relative '../interactions_test'
+
+module PacioTestKit
+  class SearchTypeTest < Inferno::Test
```

Comment:
we should have individual test file for each param and combo. since the search params and combo are the same for observation and diagnostic report, we should have a subfolder `observation_diagnostic_report` and have those test under it. we should have a tests for patient, code, category, date, patient + code,  patient + category, patient + category + date.

We can add the following as optional: status, patient + category + status, patient + code + date

Now that I think about it, let's ask the user fewer inputs as possible for better user experience. we can reuse the resource(s) retrieved from the read test to extract the value for those search parameters and use it for the search. I'll try to have the helper methods for the search test my Tuesday so that you can use them for the tests.

Reply:
Do we need to have a rspec unit test file for each search-type parameter combination, or a singular file to demonstrate one search-type case? It looks like combining all into one rspec would get confusing fast.

Reply:
I think it's better to have a unit test for each search test.

Reply:
I will push my changes sometimes today. I've implemented all search tests.  You will just need to write unit tests.

Reply:
**Update ** : 
I implemented the following search tests:
- **required**: patient, patient + code, patient + category, patient + category + date
- **optional**: patient + category + status, patient + code + date

**Correction**: The resources defined in this implementation guide (IG) are scoped specifically around the patient. As a result, any queries must be limited to resources tied to a particular patient. Allowing searches by only code, category, or date could potentially retrieve data from multiple patients, which poses a security risk. The requester may not have the necessary authorization to access data for other patients, making such searches non-compliant with privacy and security requirements. Therefore, queries must always include a patient reference to ensure that only authorized data is retrieved.